examples/platforms/da15000/radio.c

```diff
@@ -31,528 +31,463 @@
-{
-    (void)aInstance;
-    uint8_t value;
-    value = aEnable;
-    FTDF_setValue(FTDF_PIB_PROMISCUOUS_MODE, &value);
-    s_promiscuous = aEnable;
-}
+    uint8_t entry;
+    FTDF_ExtAddress addr;
+    uint32_t addrL;
+    uint64_t addrH;
 
-bool otPlatRadioGetPromiscuous(otInstance *aInstance)
-{
-    (void)aInstance;
-    return s_promiscuous;
-}
+    addrL = (aExtAddress[4] << 24) | (aExtAddress[5] << 16) | (aExtAddress[6] << 8) | (aExtAddress[7] << 0);
+    addrH = (aExtAddress[0] << 24) | (aExtAddress[1] << 16) | (aExtAddress[2] << 8) | (aExtAddress[3] << 0);
+    addr = addrL | (addrH << 32);
 
-ThreadError otPlatRadioHandleTransmitDone(bool *rxPending)
-{
+    // check if address already stored
     ThreadError error = kThreadError_None;
-    VerifyOrExit(s_state == kStateTransmit, error = kThreadError_InvalidState);
+    SuccessOrExit(FTDF_fpprLookupExtAddress(addr, &entry));
 
-    s_state = kStateReceive;
+    VerifyOrExit(FTDF_fpprGetFreeExtAddress(&entry), error = kThreadError_NoBufs);
 
-    if (rxPending != NULL)
-    {
-        *rxPending = s_data_pending;
-    }
+    FTDF_fpprSetExtAddress(entry, addr);
+    FTDF_fpprSetExtAddressValid(entry, FTDF_TRUE);
 
 exit:
     return error;
 }
 
-
-
-
-void FTDF_sendFrameTransparentConfirm(void *handle, FTDF_Bitmap32 status)
+ThreadError otPlatRadioClearSrcMatchShortEntry(otInstance *aInstance, const uint16_t aShortAddress)
 {
-    (void)handle;
-    (void)status;
-    volatile uint32_t *txStatus = FTDF_GET_REG_ADDR_INDEXED(RETENTION_RAM_TX_RETURN_STATUS_1, FTDF_TX_DATA_BUFFER);
+    (void)aInstance;
 
-    SendFrameDone = true;
+    uint8_t entry;
+    uint8_t entryIdx;
 
-#if DEBUG_LOG_ENABLE
-    hw_uart_write_buffer(HW_UART1, "\nTD", 3);
-#endif
-}
+    ThreadError error = kThreadError_None;
+    VerifyOrExit(FTDF_fpprLookupShortAddress(aShortAddress, &entry, &entryIdx), error = kThreadError_NoAddress);
 
+    FTDF_fpprSetShortAddress(entry, entryIdx, 0);
+    FTDF_fpprSetShortAddressValid(entry, entryIdx, FTDF_FALSE);
 
+exit:
+    return error;
+}
 
-void da15000RadioProcess(otInstance *aInstance)
+ThreadError otPlatRadioClearSrcMatchExtEntry(otInstance *aInstance, const uint8_t *aExtAddress)
 {
-    bool rxPending;
-    uint8_t bufferForFTDFconfigValue = 0;
+    (void)aInstance;
 
-    if (SendFrameDone)
-    {
-        FTDF_SET_FIELD(ON_OFF_REGMAP_RXENABLE, 1);
-        otPlatRadioHandleTransmitDone(&rxPending);
-        otPlatRadioTransmitDone(aInstance, &s_transmit_frame, false, kThreadError_None);
-        SendFrameDone = false;
-    }
+    uint8_t entry;
+    FTDF_ExtAddress addr;
+    uint32_t addrL;
+    uint64_t addrH;
 
-    if ((SED) && ((otPlatAlarmGetNow() - sleep_const_delay) > 100))
-    {
-        FTDF_setValue(FTDF_PIB_RX_ON_WHEN_IDLE, &bufferForFTDFconfigValue);
-        s_state = kStateSleep;
-        SED = false;
-    }
-}
+    addrL = (aExtAddress[4] << 24) | (aExtAddress[5] << 16) | (aExtAddress[6] << 8) | (aExtAddress[7] << 0);
```

Comment:
I'm not sure that two intermediary values are really needed here.  Since ``FTDF_ExtAddress`` is just a ``uint64_t``, could you not use it instead of the ``addrL`` variable?   In-fact you could do it without the ``addrH`` as well, but that may be more confusing.